src/client/metadataApiRetrieve.ts

```diff
@@ -111,6 +113,12 @@ export class RetrieveResult implements MetadataTransferResult {
       }
     }
 
+    // Add file responses for components that support partial delete (e.g., DigitalExperience)
+    // where pieces of the component were deleted in the org, then retrieved.
+    while (partialDeleteFileResponses.length) {
+      this.fileResponses.push(partialDeleteFileResponses.pop());
```

Comment:
I did it this way so that `partialDeleteFileResponses` should be empty after `getFileResponses()` is executed so as not to affect other RetrieveResults.